src/teleport.cpp

```diff
@@ -85,45 +91,47 @@ void Teleport::addThing(int32_t, Thing* thing)
 
 	const MagicEffectClasses effect = Item::items[id].magicEffect;
 
-	if (Creature* creature = thing->getCreature()) {
+	if (auto creature = thing->getCreature()) {
 		Position origPos = creature->getPosition();
 		g_game.internalCreatureTurn(creature, origPos.x > destPos.x ? DIRECTION_WEST : DIRECTION_EAST);
-		g_game.map.moveCreature(*creature, *destTile);
+		g_game.map.moveCreature(creature, destTile);
 		if (effect != CONST_ME_NONE) {
 			g_game.addMagicEffect(origPos, effect);
 			g_game.addMagicEffect(destTile->getPosition(), effect);
 		}
-	} else if (Item* item = thing->getItem()) {
+	} else if (auto item = thing->getItem()) {
```

Comment:
Could this be `const auto`
```suggestion
	} else if (const auto item = thing->getItem()) {
```